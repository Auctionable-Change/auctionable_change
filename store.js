import React, { createContext, useContext, useReducer } from "react";

const StoreContext = createContext();

const initialState = {
  currentListing: {
    id: null,
    title: "",
    description: "",
    price: null,
    donor: "",
    donor_email: "",
    status: "",
    category: "",
    charity: "",
    charity_url: "",
    charity_score: "",
    charity_score_image: "",
    image: "",
    bids: [],
  },
  buyerDetails: {
    item_id: null,
    bidder_name: "",
    bidder_email: "",
    amount: null,
    street_address: "",
    city: "",
    state: "",
    zip_code: "",
    receipt: "",
  },
  listingToPost: {},
};

const reducer = (state, action) => {
  switch (action.type) {
    case "ADD_CURRENT_LISTING":
      return {
        ...state,
        currentListing: action.currentListing,
      };
    case "ADD_SELECTED_CHARITY":
      return {
        ...state,
        listingToPost: {
          ...state.listingToPost,
          charity: action.selectedCharity.name,
          charity_url: action.selectedCharity.url,
          charity_score: action.selectedCharity.rating,
          charity_score_image: action.selectedCharity.rating_image,
        },
      };
    case "ADD_BUYER_DETAILS":
      return {
        ...state,
        buyerDetails: { ...state.buyerDetails, ...action.buyerDetails },
      };
    case "ADD_TO_LISTING":
      return {
        ...state,
        listingToPost: { ...state.listingToPost, ...action.listingToPost },
      };
    case "POST_BID":
      return {
        ...state,
        currentBid: action.bid
      };
      case "SET_USER":
      return {
        ...state,
        loginDetails: action.loginDetails,
        userInfo: action.userInfo,
        registrationDetails: action.registration
      }

    default:
      throw new Error(`Unhandled action type: ${action.type}`);
  }
};

export const StoreProvider = ({ children }) => {
  const [state, dispatch] = useReducer(reducer, initialState);

  return (
    <StoreContext.Provider value={{ state, dispatch }}>
      {children}
    </StoreContext.Provider>
  );
};

export const useStore = () => useContext(StoreContext);
