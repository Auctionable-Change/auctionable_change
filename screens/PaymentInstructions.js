import React, { useState, useEffect } from "react";
import { StyleSheet, Image, View } from "react-native";
import { ScrollView } from "react-native-gesture-handler";
import { SafeAreaView } from "react-native-safe-area-context";
import { Button, Text } from "native-base";
import { useStore } from "../store";
import * as Linking from "expo-linking";

const PaymentInstructions = ({ navigation }) => {
  const { dispatch, state } = useStore();

  return (
    <SafeAreaView style={{ flex: 1 }}>
      <ScrollView>
        <Text style={styles.title}>Thank you for your contribution!</Text>
        <Text style={styles.subheader}>Instructions to complete order:</Text>
        <Text>1. Follow this </Text>
        <Text
          style={{ fontWeight: "bold", textDecorationLine: "underline" }}
          onPress={() => Linking.openURL(state.currentListing.charity_url)}
        >
          link
        </Text>
        <Text> to donate at least ${state.currentListing.price}</Text>
        <Text>Screenshot and upload your receipt.</Text>
        {/* camera component */}
        <Button block onPress={() => navigation.navigate("Email")}>
          <Text>Continue</Text>
        </Button>
      </ScrollView>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    marginTop: 10,
  },
  subheader: {
    fontSize: 15,
    marginBottom: 20,
  },
  scroll: {
    display: "flex",
    alignItems: "center",
  },
  title: {
    fontSize: 25,
    textAlign: "center",
    margin: 10,
  },
  listItem: {
    fontSize: 15,
    margin: 5,
  },
  textInput: {
    height: 30,
    borderColor: "gray",
    borderWidth: 1,
    margin: 10,
    borderRadius: 10,
    padding: 5,
    width: 250,
  },
  button: {
    backgroundColor: "#FFFFFF",
    borderRadius: 10,
    width: 200,
    padding: 10,
    margin: 15,
    alignItems: "center",
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.18,
    shadowRadius: 1.0,
  },
  stepOne: {
    flexDirection: "row",
    flexWrap: "wrap",
    marginTop: 10,
  },
});

export default PaymentInstructions;
