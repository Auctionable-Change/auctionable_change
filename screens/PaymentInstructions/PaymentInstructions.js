import React from "react";
import { StyleSheet, View } from "react-native";
import { SafeAreaView } from "react-native-safe-area-context";
import { Text } from "native-base";
import { useStore } from "../../store";
import * as Linking from "expo-linking";
import Camera from "../Camera/Camera";

const PaymentInstructions = ({ navigation }) => {
  const { state } = useStore();

  return (
    <SafeAreaView style={styles.container}>
      <Text style={styles.title}>Thank you for your contribution!</Text>
      <Text style={styles.subtitle}>Instructions to complete order:</Text>
      <View style={styles.stepOne}>
        <Text>First, follow this </Text>
        <Text
          style={{ fontWeight: "bold", textDecorationLine: "underline" }}
          onPress={() => Linking.openURL(state.currentListing.charity_url)}
        >
          link
          {" "}
        </Text>
        <Text>to donate at least ${state.currentListing.price}.</Text>
      </View>
      <Text style={{margin: 10}}>
        Then, take a screenshot that includes your name, charity name and donation
        amount.
      </Text>
      <Text>
      Do not include other personal information.
      </Text>
      <Camera
        cameraType="launchImageLibraryAsync"
        user="buyer"
        prompt="Press camera to upload screenshot of donation receipt from your camera roll."
        title="Upload Receipt"
      />
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#FFFFFF",
    display: "flex",
    alignItems: "center",
  },
  subtitle: {
    fontSize: 15,
    margin: 5,
  },
  scroll: {
    display: "flex",
    alignItems: "center",
  },
  title: {
    fontSize: 25,
    textAlign: "center",
    margin: 5,
  },
  stepOne: {
    flexDirection: "row",
    flexWrap: "wrap",
    marginTop: 5,
  },
});

export default PaymentInstructions;
