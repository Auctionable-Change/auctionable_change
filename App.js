import React, { useEffect } from "react";
import { AppLoading } from "expo";
import * as Font from "expo-font";
import { Ionicons } from "@expo/vector-icons";
import { createStackNavigator } from "@react-navigation/stack";
import { NavigationContainer } from "@react-navigation/native";
import { StoreProvider } from "./store";
import { StyleSheet } from "react-native";
import LogoTitle from "./components/LogoTitle";
import Welcome from "./screens/Welcome/Welcome";
import CurrentListings from "./screens/CurrentListings/CurrentListings";
import ListingDetails from "./screens/ListingDetails/ListingDetails";
import PostItem from "./screens/PostItem/PostItem";
import PaymentInstructions from "./screens/PaymentInstructions/PaymentInstructions";
import ChooseCharity from "./screens/ChooseCharity/ChooseCharity";
import PostConfirmation from "./screens/PostConfirmation/PostConfirmation";
import EmailForm from "./screens/EmailForm/EmailForm";
import Camera from "./screens/Camera/Camera";
import { Image } from "react-native";

const Stack = createStackNavigator();

export default function App() {
  useEffect(() => {
    const loadFonts = async () => {
      await Font.loadAsync({
        Roboto: require("native-base/Fonts/Roboto.ttf"),
        Roboto_medium: require("native-base/Fonts/Roboto_medium.ttf"),
        ...Ionicons.font,
      });
    };
    loadFonts();
  });

  return (
    <StoreProvider>
      <NavigationContainer>
        <Stack.Navigator>
          <Stack.Screen name="Home" component={Welcome} />
          <Stack.Screen
            name="Listings"
            component={CurrentListings}
            options={{
              headerTitle: () => <LogoTitle />,
              headerBackTitleVisible: false,
              ...backButtonStyle,
            }}
          />
          <Stack.Screen
            name="Details"
            component={ListingDetails}
            options={{
              headerTitle: () => <LogoTitle />,
              headerBackTitleVisible: false,
              ...backButtonStyle,
            }}
          />
          <Stack.Screen
            name="Purchase"
            component={PaymentInstructions}
            options={{
              headerTitle: () => <LogoTitle />,
              headerBackTitleVisible: false,
              ...backButtonStyle,
            }}
          />
          <Stack.Screen
            name="Email"
            component={EmailForm}
            options={{
              headerTitle: () => <LogoTitle />,
              headerBackTitleVisible: false,
              ...backButtonStyle,
            }}
          />
          <Stack.Screen
            name="Post Item"
            component={PostItem}
            options={{
              headerTitle: () => <LogoTitle />,
              headerBackTitleVisible: false,
              ...backButtonStyle,
            }}
          />
          <Stack.Screen
            name="Choose Charity"
            component={ChooseCharity}
            options={{
              headerTitle: () => <LogoTitle />,
              headerBackTitleVisible: false,
              ...backButtonStyle,
            }}
          />
          <Stack.Screen
            name="Confirmation"
            component={PostConfirmation}
            options={{
              headerTitle: () => <LogoTitle />,
              headerBackTitleVisible: false,
              ...backButtonStyle,
            }}
          />
          <Stack.Screen
            name="Camera"
            options={{
              headerTitle: () => <LogoTitle />,
              headerBackTitleVisible: false,
              ...backButtonStyle,
            }}
          >
            {() => <Camera cameraType="launchCameraAsync"
              user="seller"
              prompt="Press camera to take a photo of your item to donate!"
              title="Upload a Photo"/>}
          </Stack.Screen>
        </Stack.Navigator>
      </NavigationContainer>
    </StoreProvider>
  );
}

const backButtonStyle = {
  headerBackImage: () => (
    <Image
      accessibilityRole="button"
      id="add"
      source={require("./assets/icons/back.png")}
      style={{ width: 30, height: 30, margin: 8 }}
    />
  )         
}