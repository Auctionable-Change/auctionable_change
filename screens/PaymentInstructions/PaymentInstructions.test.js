import React from 'react';
import { render, fireEvent, waitFor, NativeTestEvent } from 'react-native-testing-library';
import PaymentInstructions from './PaymentInstructions';
import { StoreProvider } from '../../store'
import {expect, it} from '@jest/globals';
import renderer from "react-test-renderer";
import App from '../../App'


describe('PaymentInstructions', () => {
  it('should render', () => {
    const tree = renderer.create(
          <App component={PaymentInstructions}/>
      ).toJSON();
    expect(tree).toMatchSnapshot();
  })
})