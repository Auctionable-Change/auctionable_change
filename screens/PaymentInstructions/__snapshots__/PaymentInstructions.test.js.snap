// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaymentInstructions should render 1`] = `
<View
  style={
    Object {
      "flex": 1,
    }
  }
>
  <RNCSafeAreaProvider
    onInsetsChange={[Function]}
    style={
      Array [
        Object {
          "flex": 1,
        },
        undefined,
      ]
    }
  >
    <View
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "zIndex": 1,
          },
          false,
        ]
      }
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        onLayout={[Function]}
        pointerEvents="box-none"
        style={null}
      >
        <View
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "opacity": 1,
              "position": "absolute",
              "right": 0,
              "top": 0,
              "zIndex": 0,
            }
          }
        >
          <View
            style={
              Object {
                "backgroundColor": "rgb(255, 255, 255)",
                "borderBottomColor": "rgb(216, 216, 216)",
                "flex": 1,
                "shadowColor": "rgb(216, 216, 216)",
                "shadowOffset": Object {
                  "height": 0.5,
                  "width": 0,
                },
                "shadowOpacity": 0.85,
                "shadowRadius": 0,
              }
            }
          />
        </View>
        <View
          pointerEvents="box-none"
          style={
            Object {
              "height": 64,
              "maxHeight": undefined,
              "minHeight": undefined,
              "opacity": undefined,
              "transform": undefined,
            }
          }
        >
          <View
            pointerEvents="none"
            style={
              Object {
                "height": 20,
              }
            }
          />
          <View
            pointerEvents="box-none"
            style={
              Object {
                "alignItems": "center",
                "flex": 1,
                "flexDirection": "row",
                "justifyContent": "center",
              }
            }
          >
            <View
              pointerEvents="box-none"
              style={
                Object {
                  "marginHorizontal": 16,
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityRole="header"
                numberOfLines={1}
                onLayout={[Function]}
                style={
                  Object {
                    "color": "rgb(28, 28, 30)",
                    "fontSize": 17,
                    "fontWeight": "600",
                  }
                }
              >
                Home
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      onLayout={[Function]}
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View
        pointerEvents="box-none"
        style={
          Object {
            "bottom": 0,
            "left": 0,
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          collapsable={false}
          style={
            Object {
              "opacity": 1,
            }
          }
        />
        <View
          accessibilityElementsHidden={false}
          closing={false}
          gestureVelocityImpact={0.3}
          importantForAccessibility="auto"
          onClose={[Function]}
          onGestureBegin={[Function]}
          onGestureCanceled={[Function]}
          onOpen={[Function]}
          onTransitionStart={[Function]}
          pointerEvents="box-none"
          style={
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            }
          }
          transitionSpec={
            Object {
              "close": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
              "open": Object {
                "animation": "spring",
                "config": Object {
                  "damping": 500,
                  "mass": 3,
                  "overshootClamping": true,
                  "restDisplacementThreshold": 10,
                  "restSpeedThreshold": 10,
                  "stiffness": 1000,
                },
              },
            }
          }
        >
          <View
            pointerEvents="box-none"
            style={
              Object {
                "flex": 1,
              }
            }
          >
            <View
              collapsable={false}
              forwardedRef={[Function]}
              onGestureHandlerEvent={[Function]}
              onGestureHandlerStateChange={[Function]}
              style={
                Object {
                  "flex": 1,
                  "transform": Array [
                    Object {
                      "translateX": 0,
                    },
                    Object {
                      "translateX": 0,
                    },
                  ],
                }
              }
            >
              <View
                pointerEvents="none"
                style={
                  Object {
                    "backgroundColor": "rgb(242, 242, 242)",
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "shadowColor": "#000",
                    "shadowOffset": Object {
                      "height": 1,
                      "width": -1,
                    },
                    "shadowOpacity": 0.3,
                    "shadowRadius": 5,
                    "top": 0,
                    "width": 3,
                  }
                }
              />
              <View
                style={
                  Array [
                    Object {
                      "flex": 1,
                      "overflow": "hidden",
                    },
                    Array [
                      Object {
                        "backgroundColor": "rgb(242, 242, 242)",
                      },
                      undefined,
                    ],
                  ]
                }
              >
                <View
                  style={
                    Object {
                      "flex": 1,
                      "flexDirection": "column-reverse",
                    }
                  }
                >
                  <View
                    style={
                      Object {
                        "flex": 1,
                      }
                    }
                  >
                    <View
                      style={
                        Object {
                          "flex": 1,
                        }
                      }
                    >
                      <Image
                        source={1}
                        style={
                          Object {
                            "height": 200,
                            "resizeMode": "contain",
                            "width": "95%",
                          }
                        }
                      />
                      <Text
                        style={
                          Array [
                            Object {
                              "color": "#000",
                              "fontFamily": "System",
                              "fontSize": 16,
                            },
                            Object {
                              "alignSelf": "center",
                              "color": "grey",
                              "fontStyle": "italic",
                              "marginBottom": 10,
                              "textAlign": "center",
                              "width": "70%",
                            },
                          ]
                        }
                      >
                        Buy and sell with purpose because together we can make A.change
                      </Text>
                      <View
                        style={
                          Object {
                            "flex": 1,
                            "padding": 10,
                          }
                        }
                      >
                        <View
                          accessible={true}
                          focusable={true}
                          forwardedRef={[Function]}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "alignItems": "center",
                              "alignSelf": "stretch",
                              "backgroundColor": "#62B1F6",
                              "borderBottomWidth": null,
                              "borderColor": "#007aff",
                              "borderLeftWidth": null,
                              "borderRadius": 5,
                              "borderRightWidth": null,
                              "borderTopWidth": null,
                              "borderWidth": undefined,
                              "elevation": 2,
                              "flexDirection": "row",
                              "height": 45,
                              "justifyContent": "center",
                              "opacity": 1,
                              "paddingBottom": 6,
                              "paddingTop": 6,
                              "shadowColor": undefined,
                              "shadowOffset": undefined,
                              "shadowOpacity": undefined,
                              "shadowRadius": undefined,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "backgroundColor": "transparent",
                                "color": "#fff",
                                "fontFamily": "System",
                                "fontSize": 16.5,
                                "marginLeft": 0,
                                "marginRight": 0,
                                "paddingLeft": 16,
                                "paddingRight": 16,
                              }
                            }
                          >
                            Current Listings
                          </Text>
                        </View>
                        <View
                          accessible={true}
                          focusable={true}
                          forwardedRef={[Function]}
                          onClick={[Function]}
                          onResponderGrant={[Function]}
                          onResponderMove={[Function]}
                          onResponderRelease={[Function]}
                          onResponderTerminate={[Function]}
                          onResponderTerminationRequest={[Function]}
                          onStartShouldSetResponder={[Function]}
                          style={
                            Object {
                              "alignItems": "center",
                              "alignSelf": "stretch",
                              "backgroundColor": "#62B1F6",
                              "borderBottomWidth": null,
                              "borderColor": "#007aff",
                              "borderLeftWidth": null,
                              "borderRadius": 5,
                              "borderRightWidth": null,
                              "borderTopWidth": null,
                              "borderWidth": undefined,
                              "elevation": 2,
                              "flexDirection": "row",
                              "height": 45,
                              "justifyContent": "center",
                              "marginTop": 30,
                              "opacity": 1,
                              "paddingBottom": 6,
                              "paddingTop": 6,
                              "shadowColor": undefined,
                              "shadowOffset": undefined,
                              "shadowOpacity": undefined,
                              "shadowRadius": undefined,
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "backgroundColor": "transparent",
                                "color": "#fff",
                                "fontFamily": "System",
                                "fontSize": 16.5,
                                "marginLeft": 0,
                                "marginRight": 0,
                                "paddingLeft": 16,
                                "paddingRight": 16,
                              }
                            }
                          >
                            Post Item
                          </Text>
                        </View>
                      </View>
                      <View
                        style={
                          Object {
                            "backgroundColor": "white",
                            "borderTopColor": "#dddddd",
                            "borderTopWidth": 1,
                            "bottom": "0%",
                            "display": "flex",
                            "flexDirection": "row",
                            "height": "8%",
                            "justifyContent": "space-around",
                            "padding": 8,
                            "position": "absolute",
                            "width": "100%",
                          }
                        }
                      >
                        <RNGestureHandlerButton
                          collapsable={false}
                          onGestureEvent={[Function]}
                          onGestureHandlerEvent={[Function]}
                          onGestureHandlerStateChange={[Function]}
                          onHandlerStateChange={[Function]}
                          rippleColor={0}
                          style={
                            Array [
                              Object {
                                "overflow": "hidden",
                              },
                              undefined,
                            ]
                          }
                        >
                          <View
                            accessible={true}
                          >
                            <Image
                              accessibilityRole="button"
                              id="search"
                              source={1}
                              style={
                                Object {
                                  "height": 30,
                                  "width": 30,
                                }
                              }
                            />
                          </View>
                        </RNGestureHandlerButton>
                        <RNGestureHandlerButton
                          collapsable={false}
                          onGestureEvent={[Function]}
                          onGestureHandlerEvent={[Function]}
                          onGestureHandlerStateChange={[Function]}
                          onHandlerStateChange={[Function]}
                          rippleColor={0}
                          style={
                            Array [
                              Object {
                                "overflow": "hidden",
                              },
                              undefined,
                            ]
                          }
                        >
                          <View
                            accessible={true}
                          >
                            <Image
                              accessibilityRole="button"
                              id="add"
                              source={1}
                              style={
                                Object {
                                  "height": 30,
                                  "width": 30,
                                }
                              }
                            />
                          </View>
                        </RNGestureHandlerButton>
                        <RNGestureHandlerButton
                          collapsable={false}
                          onGestureEvent={[Function]}
                          onGestureHandlerEvent={[Function]}
                          onGestureHandlerStateChange={[Function]}
                          onHandlerStateChange={[Function]}
                          rippleColor={0}
                          style={
                            Array [
                              Object {
                                "overflow": "hidden",
                              },
                              undefined,
                            ]
                          }
                        >
                          <View
                            accessible={true}
                          >
                            <Image
                              accessibilityRole="button"
                              id="home"
                              source={1}
                              style={
                                Object {
                                  "height": 30,
                                  "width": 30,
                                }
                              }
                            />
                          </View>
                        </RNGestureHandlerButton>
                      </View>
                    </View>
                  </View>
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </RNCSafeAreaProvider>
</View>
`;
